test/capture_udp.erl

```diff
@@ -74,6 +75,19 @@ peek() ->
 read() ->
     gen_server:call(?SERVER, read).
 
+-spec read_at_least(non_neg_integer()) -> {non_neg_integer(), iolist()}.
+read_at_least(Num) ->
```

Comment:
So this is nice to avoid the sleep. You could tweak it as follows:
Make a /2 helper like `read_at_least(N, Acc)`. You then have a fun head with `read_at_least(0, Acc)` that will not recurse but return Acc. You could also slightly improve it by building a deep list and flattening it (maybe reversing too) at the end. For these purposes it is fine as is, but generally better to accumulate with cons and then post-process rather than list append in the recursive loop. Make sense?